README.md

```diff
@@ -136,43 +151,55 @@ const connector = new AppSearchAPIConnector({
 
 Search UI can connect to **any** web based Search API. Read [the advanced README](./ADVANCED.md#build-your-own-connector) for more information.
 
-### 2. SearchProvider
+### 3. SearchProvider
 
-The `SearchProvider` object will tie all of your Components together so that they work as a cohesive application.
+`SearchProvider` is the top level component in your Search UI implementation.
+It is where you configure your search experience, and it ties all of your components together so that they work as a cohesive application.
 
-It's where you configure your UI, acting as the state manager between "State" and "Actions".
+`SearchProvider` can be used with components:
```

Comment:
I love the new content in this section. But I think we can smooth the flow. 🌊 

---

SearchProvider can be used with components:

```
<SearchProvider
  config={{
    apiConnector: connector
  }}
>
  {() => <div className="App">{/* Place Components here! */}</div>}
</SearchProvider>

```

---

The idea that you can use searchProvider without components is important. 

But the fact that you _can_ use it with components should lead into the Component section.

ie.

...

You can use searchProvider directly with state and actions:

```
<SearchProvider
  config={{
    apiConnector: connector
  }}
>
  {({ searchTerm, setSearchTerm }) => (
    <div className="App">{/* Work directly with state and actions! */}</div>
  )}
</SearchProvider>
```

One way to think of `SearchProvider` is as a search experience without a view. Instead of thinking of a search experience in terms of UI like search boxes and dropdowns, it thinks of the underlying "state" and "actions" associated with a search experience. So instead of a "search box", the `SearchProvider` just cares about the underlying `searchTerm` that is populated in a search box (the state), and the action that a user takes on that search box when they change the `searchTerm`, i.e, `setSearchTerm` (the action).

**Since you are able to work directly with "state" and "actions", you can use any type of input you want!** As long as it ends up calling `setSearchTerm`, it will "just work". This gives you maximum flexbility over your experience.

**But there are also pre-built search features consumed by searchProvider. You can quickly add a searchBox, etc, etc. They're called Components.**

_Next section_

---

That way, you introduce that tasty, flexibly awesome searchProvider idea _and then_ the component idea. ❤️

Reply:
I agree, the flow wasn't great before. I adjusted it, but kind of in my own way: https://github.com/elastic/search-ui/pull/226/commits/f87a0582be2cc60433cda03f066eaf7461ddd790.

Let me know what you think.

The verbiage explaining "state" and "actions" more in depth I'm going to move to the Advanced Headless Core guide.

Reply:
👍